File: Leetcode/Easy/746. Min Cost Climbing Stairs.cpp
class Solution
{
public:
    int minCostClimbingStairs(vector<int> &cost)
    {
        int n = cost.size();
        int res = 0;
        int prev = 0;
        int second = 0;
        for (int i = 2; i < n + 1; i++)
        {
            res = min(prev + cost[i - 1], second + cost[i - 2]);
            second = prev;
            prev = res;
        }
        return res;
    }
};

File: Leetcode/Easy/746. Min Cost Climbing Stairs/746. Min Cost Climbing Stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        memo = {}
        def min_cost(i):
            if i <= 1:
                return 0

            if i in memo:
                return memo[i]
            
            memo[i] = min(cost[i - 1] + min_cost(i - 1), cost[i - 2] + min_cost(i - 2))

            return memo[i]
        return min_cost(len(cost)) 